cmd/gazelle/update-repos.go

```diff
@@ -108,20 +131,34 @@ func updateRepos(args []string) error {
 	}
 	uc := getUpdateReposConfig(c)
 
-	workspacePath := filepath.Join(c.RepoRoot, "WORKSPACE")
-	f, err := rule.LoadWorkspaceFile(workspacePath, "")
+	var f *rule.File
+	var path string
+	if uc.macroFileName == "" {
+		path = filepath.Join(c.RepoRoot, "WORKSPACE")
+		f, err = rule.LoadWorkspaceFile(path, "")
+	} else {
+		path = uc.macroFileName
+		f, err = rule.LoadMacroFile(path, "", uc.macroDefName)
+	}
 	if err != nil {
-		return fmt.Errorf("error loading %q: %v", workspacePath, err)
+		return fmt.Errorf("error loading %q: %v", path, err)
+	}
+
+	if uc.macroFileName == "" {
+		merger.FixWorkspace(f)
 	}
-	merger.FixWorkspace(f)
 
 	if err := uc.fn(uc, f, kinds); err != nil {
 		return err
 	}
-	merger.FixLoads(f, loads)
-	if err := merger.CheckGazelleLoaded(f); err != nil {
-		return err
+
+	if uc.macroFileName == "" {
+		merger.FixLoads(f, loads)
```

Comment:
`FixLoads` should still work for macro files. That will require some changes in `File.Sync`, since the loads need to be outside of the macro.

Reply:
Fixed and tested with TestImportReposFromDepToEmptyMacro